scripts/zkai_native_seq32_attention_mlp_single_proof_gate.py

```
@@ -0,0 +1,684 @@
+    if summary["native_single_proof_typed_bytes"] >= summary["matched_two_proof_frontier_typed_bytes"]:
+        raise NativeSeq32AttentionMlpSingleProofGateError("typed saving missing")
+    if summary["typed_saving_vs_matched_frontier_bytes"] != TYPED_SAVING_BYTES:
+        raise NativeSeq32AttentionMlpSingleProofGateError("typed saving drift")
+    if summary["proof_size_comparable_external_rows"] != 0:
+        raise NativeSeq32AttentionMlpSingleProofGateError("external comparison overclaim")
+    if payload.get("payload_commitment") != payload_commitment(payload):
+        raise NativeSeq32AttentionMlpSingleProofGateError("payload commitment drift")
+    if "mutation_result" in payload:
+        result = _dict(payload["mutation_result"], "mutation result")
+        if not result.get("all_mutations_rejected"):
+            raise NativeSeq32AttentionMlpSingleProofGateError("mutation result drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)